sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -293,9 +293,9 @@ def test_piecewise_solve():
 
     g = Piecewise(((x - 5)**5, x >= 2), (f, x < 2))
     assert solve(g, x) == [5]
-
+    # solve returing {2,5} for below.
     g = Piecewise(((x - 5)**5, x >= 2), (f, True))
-    assert solve(g, x) == [5]
```

Comment:
solveset is bug free in many cases, so I thought we can replace `solver` with `solveset`, if solveset is returning correct answer for particular test, rather than fixing the problems in `solver`.
Now I should try to fix this for `solver`.